### PR TITLE
config reload fixes

### PR DESCRIPTION
**Pull request by sallyom**
* cleanup and a fixes for config reloading
* remove the defaultHostConfig and the filetransfer when processing the config  - I was wrong about that - we don't need this, since the config file will be on the host wherever the mount point of :/opt/mount passed to the podman command is. I dunno what I was thinking, I think I was tired.

### CODE REVIEW

File: pkg/engine/harpoon.go [hunk 1]
```diff
@@ -105,10 +104,8 @@ func (o *HarpoonConfig) bindFlags(cmd *cobra.Command) {
 // new targets will be added, stale removed, and existing
 // will set last commit as last known.
 func (hc *HarpoonConfig) Restart() {
-	if hc.scheduler == nil {
-		return
-	}
 	hc.scheduler.RemoveByTags(kubeMethod, ansibleMethod, fileTransferMethod, systemdMethod, rawMethod)
```
Comment (sallyom) [inline pkg/engine/harpoon.go:107]: the scheduler will never be nil here, this is not needed

File: pkg/engine/harpoon.go [hunk 2]
```diff
@@ -381,16 +377,6 @@ func (hc *HarpoonConfig) processConfig(ctx context.Context, target *Target) {
...
-	mo := &SingleMethodObj{
-		Conn:   hc.conn,
-		Method: configMethod,
-		Target: target,
-	}
-	if err := hc.EngineMethod(ctx, mo, defaultConfigPath, nil); err != nil {
-		klog.Warningf("Target: %s error updating config targets, will retry next run: %v", err)
-		return
-	}
-
```
Comment (sallyom) [inline pkg/engine/harpoon.go:393]: this filetransfer is not needed - whatever is passed to podman at /path/to/configdir:/opt/mount is where the config & backup config will be

File: pkg/engine/helpers.go [hunk 3]
```diff
@@ -37,39 +37,35 @@ func downloadUpdateConfigFile(urlStr string, existsAlready, initial bool) (bool,
...
-		return true, nil
-	}
-	currentConfigBytes, err := ioutil.ReadFile(defaultConfigPath)
-	if err != nil {
-		klog.Infof("unable to read current config, will try with new downloaded config file: %v", err)
-		existsAlready = false
-	} else {
-		if bytes.Equal(newBytes, currentConfigBytes) {
-			return false, nil
+	if newBytes == nil {
```
Comment (sallyom) [inline pkg/engine/helpers.go:40]: that needed a cleanup, and was broken :( makes much more sense now - again, tired.

File: pkg/engine/harpoon.go [hunk 4]
```diff
@@ -775,12 +761,6 @@ func (hc *HarpoonConfig) findDiff(target *Target, method, targetPath string, com
 // Each engineMethod call now owns the prev and dest variables instead of being shared in mo
 func (hc *HarpoonConfig) EngineMethod(ctx context.Context, mo *SingleMethodObj, path string, change *object.Change) error {
 	switch mo.Method {
-	case configMethod:
-		// Only here if config has been updated by CheckForConfigUpdates.
-		// If so, update files on disk with fileTransferPodman.
-		// Updated config file is at /opt/mount/config.yaml in harpoon pod
-		// cp updated config /opt/mount/config.yaml in pod to $HOME/.harpoon/config.yaml on host
-		return fileTransferPodman(ctx, mo, defaultConfigPath, defaultHostConfigPath, nil)
```
Comment (sallyom) [inline pkg/engine/harpoon.go:783]: this filetransfer is not needed - whatever is passed to podman at /path/to/configdir:/opt/mount is where the config & backup config will be on the host (duh)

Comment (sallyom) [reply pkg/engine/harpoon.go:783]: (i give myself permission to be snarky to myself)

**Review by cooktheryan — APPROVED** [review]

**Pull request merged**